diff --git a/src/Domain/Entity/Event.php b/src/Domain/Entity/Event.php
index f70bf18..74e44c4 100644
--- a/src/Domain/Entity/Event.php
+++ b/src/Domain/Entity/Event.php
@@ -55,7 +55,7 @@ class Event
      */
     private array $categories = [];
 
-    public function __construct(UniqueIdentifier $uniqueIdentifier = null)
+    public function __construct(?UniqueIdentifier $uniqueIdentifier = null)
     {
         $this->uniqueIdentifier = $uniqueIdentifier ?? UniqueIdentifier::createRandom();
         $this->touchedAt = new Timestamp();
@@ -71,7 +71,7 @@ class Event
         return $this->touchedAt;
     }
 
-    public function touch(Timestamp $dateTime = null): self
+    public function touch(?Timestamp $dateTime = null): self
     {
         $this->touchedAt = $dateTime ?? new Timestamp();
 
diff --git a/src/Domain/Entity/TimeZone.php b/src/Domain/Entity/TimeZone.php
index fe54108..7d0d18f 100644
--- a/src/Domain/Entity/TimeZone.php
+++ b/src/Domain/Entity/TimeZone.php
@@ -34,8 +34,8 @@ class TimeZone
 
     public static function createFromPhpDateTimeZone(
         PhpDateTimeZone $phpDateTimeZone,
-        DateTimeInterface $beginDateTime = null,
-        DateTimeInterface $endDateTime = null
+        ?DateTimeInterface $beginDateTime = null,
+        ?DateTimeInterface $endDateTime = null
     ): self {
         if ($beginDateTime === null || $endDateTime === null) {
             trigger_deprecation('eluceo/ical', '2.1.0', 'Relying on the default values for begin and end date when calling TimeZone::createFromPhpDateTimeZone() is deprecated. Please provide a begin and an end date.');
diff --git a/src/Domain/ValueObject/Attachment.php b/src/Domain/ValueObject/Attachment.php
index 1f230ec..5192011 100644
--- a/src/Domain/ValueObject/Attachment.php
+++ b/src/Domain/ValueObject/Attachment.php
@@ -22,7 +22,7 @@ class Attachment
     /**
      * @param BinaryContent|Uri $content
      */
-    public function __construct(object $content, string $mimeType = null)
+    public function __construct(object $content, ?string $mimeType = null)
     {
         $this->mimeType = $mimeType;
 
diff --git a/src/Domain/ValueObject/Location.php b/src/Domain/ValueObject/Location.php
index 2f23729..9680392 100644
--- a/src/Domain/ValueObject/Location.php
+++ b/src/Domain/ValueObject/Location.php
@@ -17,7 +17,7 @@ final class Location
     private ?string $title;
     private ?GeographicPosition $geographicPosition = null;
 
-    public function __construct(string $location, string $title = null)
+    public function __construct(string $location, ?string $title = null)
     {
         $this->location = $location;
         $this->title = $title;
diff --git a/src/Domain/ValueObject/Organizer.php b/src/Domain/ValueObject/Organizer.php
index c3948ee..0989bf8 100644
--- a/src/Domain/ValueObject/Organizer.php
+++ b/src/Domain/ValueObject/Organizer.php
@@ -30,9 +30,9 @@ final class Organizer
 
     public function __construct(
         EmailAddress $emailAddress,
-        string $displayName = null,
-        Uri $directoryEntry = null,
-        EmailAddress $sentBy = null
+        ?string $displayName = null,
+        ?Uri $directoryEntry = null,
+        ?EmailAddress $sentBy = null
     ) {
         $this->emailAddress = $emailAddress;
         $this->displayName = $displayName;
diff --git a/src/Domain/ValueObject/PointInTime.php b/src/Domain/ValueObject/PointInTime.php
index b68a009..cb24310 100644
--- a/src/Domain/ValueObject/PointInTime.php
+++ b/src/Domain/ValueObject/PointInTime.php
@@ -23,7 +23,7 @@ abstract class PointInTime
 {
     private PhpDateTimeImmutable $dateTime;
 
-    public function __construct(PhpDateTimeInterface $dateTime = null)
+    public function __construct(?PhpDateTimeInterface $dateTime = null)
     {
         if ($dateTime === null) {
             $dateTime = new PhpDateTimeImmutable();
diff --git a/src/Presentation/Factory/CalendarFactory.php b/src/Presentation/Factory/CalendarFactory.php
index be439d3..ed4e584 100644
--- a/src/Presentation/Factory/CalendarFactory.php
+++ b/src/Presentation/Factory/CalendarFactory.php
@@ -23,7 +23,7 @@ class CalendarFactory
     private EventFactory $eventFactory;
     private TimeZoneFactory $timeZoneFactory;
 
-    public function __construct(EventFactory $eventFactory = null, TimeZoneFactory $timeZoneFactory = null)
+    public function __construct(?EventFactory $eventFactory = null, ?TimeZoneFactory $timeZoneFactory = null)
     {
         $this->eventFactory = $eventFactory ?? new EventFactory();
         $this->timeZoneFactory = $timeZoneFactory ?? new TimeZoneFactory();
diff --git a/src/Presentation/Factory/EventFactory.php b/src/Presentation/Factory/EventFactory.php
index 820ecaa..21817ad 100644
--- a/src/Presentation/Factory/EventFactory.php
+++ b/src/Presentation/Factory/EventFactory.php
@@ -46,7 +46,7 @@ class EventFactory
     private DateTimeFactory $dateTimeFactory;
     private AttendeeFactory $attendeeFactory;
 
-    public function __construct(AlarmFactory $alarmFactory = null, DateTimeFactory $dateTimeFactory = null, AttendeeFactory $attendeeFactory = null)
+    public function __construct(?AlarmFactory $alarmFactory = null, ?DateTimeFactory $dateTimeFactory = null, ?AttendeeFactory $attendeeFactory = null)
     {
         $this->alarmFactory = $alarmFactory ?? new AlarmFactory();
         $this->dateTimeFactory = $dateTimeFactory ?? new DateTimeFactory();
